tests/unit/test-firefox/test.preferences.js

```diff
@@ -34,4 +36,54 @@ describe('firefox/preferences', () => {
 
   });
 
+  describe('coerceCLICustomPreference', () => {
+
+    it('converts a single --pref cli option from string to object', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=true');
+      assert.isObject(prefs);
+      assert.equal(prefs['valid.preference'], true);
+    });
+
+    it('converts array of --pref cli option values into object', () => {
+      const prefs = coerceCLICustomPreference([
+        'valid.preference=true', 'valid.preference2=false',
+      ]);
+      assert.isObject(prefs);
+      assert.equal(prefs['valid.preference'], true);
+      assert.equal(prefs['valid.preference2'], false);
+    });
+
+    it('converts boolean values', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=true');
+      assert.equal(prefs['valid.preference'], true);
+    });
+
+    it('converts number values', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=455');
+      assert.equal(typeof prefs['valid.preference'], 'number');
+    });
+
+    it('converts float values', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=4.55');
+      assert.equal(prefs['valid.preference'], '4.55');
+    });
+
+    it('does not allow certain default preferences to be customized', () => {
+      const nonChangeablePrefs = nonOverridablePreferences.map((prop) => {
+        return prop += '=true';
+      });
+      const prefs = coerceCLICustomPreference(nonChangeablePrefs);
+      for (var pref of nonChangeablePrefs) {
+        assert.equal(typeof(prefs[pref]), 'undefined');
```

Comment:
We should change this assertion so that it use the specific chai assertion and it prints an helpful message if it fails:
```
assert.isUndefined(prefs[pref], `${pref} should be undefined`);
```